include/tvm/tir/sparse.h

```diff
@@ -99,23 +99,48 @@ class DenseAxis : public Axis {
   TVM_DEFINE_OBJECT_REF_METHODS(DenseAxis, Axis, DenseAxisNode);
 };
 
+/*!
+ * \brief Sparse axis whose column indices is not consecutive.
+ */
+class SparseAxisNode : public AxisNode {
+ public:
+  static constexpr const char* _type_key = "tir.sparse.SparseAxis";
+  TVM_DECLARE_BASE_OBJECT_INFO(SparseAxisNode, AxisNode);
+};
+
+/*!
+ * \brief Managed reference to SparseAxisNode.
+ * \sa SparseAxisNode
+ */
+class SparseAxis : public Axis {
+ public:
+  TVM_DEFINE_OBJECT_REF_METHODS(SparseAxis, Axis, SparseAxisNode);
+};
+
 /*!
  * \brief Dense axis with fixed length per row.
  */
 class DenseFixedAxisNode : public DenseAxisNode {
  public:
+  Optional<SparseAxis> from_sparse;
```

Comment:
Suppose we have a sparse axis `I`. As you know, when defining a SparseBuffer we provide the list of axes of the buffer. And sometimes there is a `T.to_dense(I)` among those axis. The following is an example:
https://github.com/yzh119/tvm/blob/fb2c49fd9684c87d1b922e66784997cac170bb1a/tests/python/unittest/test_tir_sparse_script_roundtrip.py#L33

The fact is, without knowing which sparse axis was a dense-fixed axis created from, we can never print out `T.to_dense(...)` from a PrimFunc. And instead the printer will print a new `T.dense_fixed(...)` for that axis. This behavior causes inconsistency between the original script and the newly printed script.

Therefore, for the print purpose, the `from_sparse` field is introduced. It might seem an unnecessary stuff. And I haven't tested whether removing it will influence the roundtrip property (may or may not). Anyway, it doesn't hurt, and this is why I added this.

Reply:
So the major purpose is for printing.
Yeah another idea I can come up with is to store a relation graph among these axes.

Reply:
It's doable. Just wonder whether the graph has other usage? IIUC there is only one kind of relation indicated by `to_dense(...)` (is this true?), and the graph might be unnecessary for us 🤔.

Reply:
Haha I'm not sure, yeah I agree that it's not necessary at this stage.